src/main/java/pl/uj/io/cuteanimals/model/action/GoAction.java

```diff
@@ -16,30 +19,44 @@
  * @since 0.0.1-SNAPSHOT
  */
 public class GoAction implements IAction {
-    private List<String> where;
+    private final Map<String, ILocation> locations;
+    private List<String> args;
 
-    public GoAction() {
-        this.where = new ArrayList<>();
-    }
-
-    public GoAction(List<String> where) {
-        this.where = where;
+    public GoAction(Map<String, ILocation> wheres) {
+        locations = wheres;
+        args = new ArrayList<>();
     }
 
     @Override
     public List<String> getArgs() {
-        return where;
+        return args;
     }
 
     @Override
-    public void setArgs(List<String> where) {
-        this.where = where;
+    public void setArgs(List<String> args) {
+        this.args = args;
     }
 
     @Override
     public IResult execute(ICharacter character) {
-        // TODO: fetch location object matching `where` and move player there,
-        //  if location does not exist return suitable message
-        return new Result("Going to " + where.toString());
+        if (!getAcceptableStates().contains(character.getCurrentGameState())) {
+            return new Result("This action cannot be executed now");
```

Comment:
Maybe change to sth like:
```suggestion
            return new Result("You can't go somewhere else right now");
```
Just to keep immersion.